### PR TITLE
fix(cluster-create): A 404 can happen right after the cluster creatio…

### PR DESCRIPTION
**Pull request by th3morg**
…n starts. Tolerate a few 404s from the API before giving up

## Description

The API tests using this feature sometimes fail because of a 404 call to `GET /cluster/:clusterId`. This seems to be the fault of us calling EC2 to get the status of the k8s cluster. There can be a briefly period where it isn't found. Let's tolerate a few 404s before considering the cluster creation a failure.  CDAAS-2228 for additional details.
## Motivation and Context

Fix flaky API test, but also help prevent a user from actually seeing this issue.

## How has this been tested? How can a reviewer test these changes?

New automated test and manual execution to ensure non-break change
# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [ ] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [x] Have you verified the specific change made in this PR?

### CODE REVIEW

File: cmd/cluster/cluster_create.go [hunk 1]
```diff
@@ -96,8 +96,17 @@ func (o *CreateOptions) Run(cmd *cobra.Command) error {
 	o.InitializeProgressBar(cmd.OutOrStdout())
 	o.saveData.setAgentIdentifier(createSandboxRequest.AgentIdentifier)
 	o.saveData.setCreateSandboxResponse(*sandboxResponse)
+	var configErr *configuration.ConfigError
+	attempts := 0
 	for {
 		cluster, err := o.ArmoryClient.Sandbox().Get(ctx, o.saveData.getClusterId())
+		if errors.As(err, &configErr) && attempts < 3 {
```
Comment (fieldju) [inline cmd/cluster/cluster_create.go]: I thought we were using: https://github.com/eapache/go-resiliency or https://github.com/hashicorp/go-retryablehttp in our code bases.

Comment (fieldju) [reply cmd/cluster/cluster_create.go]: https://github.com/armory-io/armory-cli/blob/main/pkg/armoryCloud/client.go#L39

:point_up: could possibly be an instance of :point_down: 

https://github.com/armory-io/go-commons/blob/c916a06372fc1f8b357871e6159ed0a084c1ea75/wormhole/proxy/wormhole.go#L51-L58

**Review by fieldju — APPROVED** [review]

File: cmd/cluster/cluster_create.go (hunk 1 above)
Comment (th3morg) [reply cmd/cluster/cluster_create.go]: Yes, yes. You are right. I was being dense. Thank you for the helpful links!

Comment (th3morg) [reply cmd/cluster/cluster_create.go]: All set

**Pull request merged**